### ISSUE TITLE
CWD

### ISSUE THREAD

**Issue by ctrueden** [scijava/jaunch#20]
On OS X 10.9 Mavericks (and maybe later versions of macOS?), the PWD variable might not be set, in which case we need to chdir ourselves to the application directory, or else the cwd will be / when double-clicked from Finder. TEST THIS.

Relatedly, on pre-Mavericks, the added `-psn_` arg messes things up. But we probably don't need to worry about this in 2024.

**Comment by ctrueden** [comment]
I tested on an Apple M2 running macOS Sonoma 14.6, and indeed, when the .app is double-clicked, the CWD for the program is `/`. (I tested using `System.getProperty("user.dir")` in Fiji's Script Editor in Groovy mode.)

The question is: does it matter? Why does the CWD need to match the `${app-dir}`?

I also tested on Ubuntu Linux by launching Fiji from the Apps search. My Fiji is defined in `/home/curtis/.local/share/applications/Fiji.desktop`, pointing to `/home/curtis/Apps/Fiji/fiji-linux-x64` for the executable. When launching in this manner, the CWD ends up as `/home/curtis`—i.e. also not `${app-dir}`. 🤷

**Comment by marktsuchida** [comment]
Some notes:
- The `open` command on macOS does the same as double-clicking (sets the cwd to `/`)
- When directly invoking the launcher executable (within the `.app` bundle or otherwise) from the shell, the cwd is the shell's cwd (of course); in this case it is not necessarily clear that setting the cwd to `${app-dir}` is the desired behavior (probably depends on the app)
- On Windows, double-clicked executables get the directory containing the `.exe` as cwd

**Comment by ctrueden** [comment]
Yep, from [the documentation](https://specifications.freedesktop.org/desktop-entry-spec/latest/recognized-keys.html):

> Path - If entry is of type Application, the working directory to run the program in. 

So really, the user's home directory is only the default, meaning we don't have to special case anything in Jaunch for this.

**Comment by marktsuchida** [comment]
I think that would be reasonable for macOS.

For Linux `.desktop`, it looks like the `Exec` field honors `$PATH` while the `Path` field [needs](https://www.mail-archive.com/xdg@lists.freedesktop.org/msg08446.html) to be a static absolute path. I guess one could set `Path=/` and do the same thing as on macOS, though whether having that be the only option is a good idea I'm not sure.

There may be apps that want the CWD to be `$HOME` on all platforms (for example, because it was first developed on Linux), but I don't know if that is worth considering until it arises.
